custom_model_runner/datarobot_drum/drum/root_predictors/predict_mixin.py

```diff
@@ -368,19 +368,42 @@ def do_predict_unstructured(self, logger=None):
         return response, response_status
 
     def do_chat(self, logger=None):
+        unsupported_chat_message = (
+            "This model's chat interface was called, but chat is not supported."
+        )
+        undefined_chat_message = (
+            "This model's chat interface was called, but chat() is not implemented."
+        )
+        # _predictor is a BaseLanguagePredictor attribute of PredictionServer;
+        # see PredictionServer.__init__()
+        if not self._predictor.supports_chat():
+            if logger is not None:
+                logger.error(unsupported_chat_message)
+            return (
+                {"message": "ERROR: " + unsupported_chat_message},
+                HTTP_422_UNPROCESSABLE_ENTITY,
+            )
+
         completion_create_params = request.json
+        try:
+            result = self._predictor.chat(completion_create_params)
+            if not is_streaming_response(result):
+                response = result.to_dict()
+            else:
+                response = Response(
+                    stream_with_context(PredictMixin._stream_openai_chunks(result)),
+                    mimetype="text/event-stream",
+                )
 
-        result = self._predictor.chat(completion_create_params)
-        if not is_streaming_response(result):
-            response = result.to_dict()
-        else:
-            response = Response(
-                stream_with_context(PredictMixin._stream_openai_chunks(result)),
-                mimetype="text/event-stream",
+            return response, HTTP_200_OK
+        except TypeError:  # if chat() is not defined, hook evaluates to None
+            if logger is not None:
```

Comment:
nit: can we move this into a method? So many `if`s just to check if logger is defined or not makes code unreadable.  Something like `def log_if_possible()` :)